api/config.go

```diff
@@ -12,14 +24,128 @@ type ConfigUnset struct {
 
 // Config is the structure of an app's config.
 type Config struct {
-	Owner    string                 `json:"owner,omitempty"`
-	App      string                 `json:"app,omitempty"`
-	Values   map[string]interface{} `json:"values,omitempty"`
-	Memory   map[string]interface{} `json:"memory,omitempty"`
-	CPU      map[string]interface{} `json:"cpu,omitempty"`
-	Tags     map[string]interface{} `json:"tags,omitempty"`
-	Registry map[string]interface{} `json:"registry,omitempty"`
-	Created  string                 `json:"created,omitempty"`
-	Updated  string                 `json:"updated,omitempty"`
-	UUID     string                 `json:"uuid,omitempty"`
+	Owner        string                 `json:"owner,omitempty"`
+	App          string                 `json:"app,omitempty"`
+	Values       map[string]interface{} `json:"values,omitempty"`
+	Memory       map[string]interface{} `json:"memory,omitempty"`
+	CPU          map[string]interface{} `json:"cpu,omitempty"`
+	Healthcheck  map[string]Healthcheck `json:"healthcheck,omitempty"`
+	Tags         map[string]interface{} `json:"tags,omitempty"`
+	Registry     map[string]interface{} `json:"registry,omitempty"`
+	Created      string                 `json:"created,omitempty"`
+	Updated      string                 `json:"updated,omitempty"`
+	UUID         string                 `json:"uuid,omitempty"`
+}
+
+type Healthcheck interface {
+	fmt.Stringer
+	Probe() fmt.Stringer
+}
+
+type KVPair struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type ExecProbe struct {
+	Command []string `json:"command"`
+}
+
+func (e ExecProbe) String() string {
+	msg := "Command: "
+	for _, cmd := range e.Command {
+		msg += cmd + " "
+	}
+	return strings.TrimSpace(msg)
+}
+
+type HTTPGetProbe struct {
+	Path        string   `json:"path"`
+	Port        string   `json:"port"`
+	HTTPHeaders []*KVPair `json:"httpHeaders"`
+}
+
+func (h HTTPGetProbe) String() string {
+	return fmt.Sprintf(`Port: %s
+Path: %s
+HTTP Headers: %s`, h.Port,
+		h.Path,
+		h.HTTPHeaders)
+}
+
+type TCPSocketProbe struct {
+	Port string `json:"port"`
+}
+
+func (t TCPSocketProbe) String() string {
+	return fmt.Sprintf("Port: %s", t.Port)
+}
+
+type HealthcheckHTTPGetProbe struct {
+	InitialDelaySeconds string        `json:"initialDelaySeconds"`
+	TimeoutSeconds      string        `json:"timeoutSeconds"`
+	PeriodSeconds       string        `json:"periodSeconds"`
+	SuccessThreshold    string        `json:"successThreshold"`
+	FailureThreshold    string        `json:"failureThreshold"`
```

Comment:
All these should be integers.
http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe